src/sardana/spock/spockms.py

```diff
@@ -570,7 +570,12 @@ def _addElement(self, element_data):
             # TODO: when it becomes possible to do:
             # some taurus.Device.<attr name> = <value>
             # replace device_proxy with element
-            device_proxy = element.getObj().getHWObj()
+            try:
+                # Taurus4 usage
+                device_proxy = element.getObj().getDeviceProxy()
+            except:
```

Comment:
I wouldn't use such a broad `except` clause since it might overshadow some other exceptions happening inside `getDeviceProxy`. I'd use `except AttributeError` - AFAIK that's what's going to appear here with Taurus 3.